metadata-ingestion/src/datahub/ingestion/source/state_provider/datahub_ingestion_checkpointing_provider.py

```diff
@@ -17,15 +17,25 @@
 
 
 class DatahubIngestionStateProviderConfig(IngestionCheckpointingProviderConfig):
-    datahub_api: Optional[DatahubClientConfig] = DatahubClientConfig()
+    datahub_api: DatahubClientConfig = DatahubClientConfig()
```

Comment:
this should remain optional

Reply:
There is no use of making it optional. It will just add the extra useless if condition of checking datahub_api because of lint error.
Still if it needs to remain optional let me know.